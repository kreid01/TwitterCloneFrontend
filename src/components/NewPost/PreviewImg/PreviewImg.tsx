import React from 'react';

interface Props {
    imgSrc:string
}

export const PreviewImg: React.FC<Props> = ({imgSrc}) => {

    return (
        <img alt='' 
        data-testid='previewImg'  
        className='w-80 pt-2 pb-6 max-h-72 rounded-xl'
         src={imgSrc}/>
    )
}