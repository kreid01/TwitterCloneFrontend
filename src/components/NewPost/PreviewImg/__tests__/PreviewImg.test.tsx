import { screen, render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import renderer from 'react-test-renderer'


import { PreviewImg } from '../PreviewImg';

const mockImgUrl = ''

afterEach(cleanup)
it("renders an image when user inputs", () => {
    render(
        <PreviewImg
        imgSrc={mockImgUrl}/>) 
    expect(screen.getByTestId("previewImg")).toBeVisible()
})

it("matched snapshot", () => {
    const tree = renderer.create(<PreviewImg imgSrc={mockImgUrl}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
