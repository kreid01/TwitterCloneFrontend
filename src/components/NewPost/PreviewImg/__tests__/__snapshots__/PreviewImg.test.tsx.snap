// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<img
  alt=""
  className="w-80 pt-2 pb-6 max-h-72 rounded-xl"
  data-testid="previewImg"
  src=""
/>
`;
